### PR TITLE
Include description and hint, and switching between those names when needed

### PR DESCRIPTION
**Pull request by calderon**
### What is the goal?

Right now we are using description, hint or title without following a base. This try to keep consistence into components.

- [ ] It passses the `jt-frontend` lint commands (`--lint-js`, `--lint-less`, `--lint-coffee`, ...)
- [ ] The `README.md` has been updated accordingly
- [ ] **Issue:** (small PR's don't need it)

### How is being implemented?

We have modified some fragments and included more selectors to some of them.

### Reviewers

@jobandtalent/frontend @tomasperezv

### CODE REVIEW

**Comment by scumah** [comment]
🆗

**Comment by montiu** [comment]
👍

File: src/javascript/dom/builder/fragment/date.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ class DateField extends Base {
       },
       {
         selector: '.dynamic_form_hint',
-        value: options.data.description
+        value: options.data.hint
```
Comment (tomasperezv) [inline src/javascript/dom/builder/fragment/date.js]: Can't be `hint`, this should be `title`, as in:

```
{
  "id": 1065,
  "key": "birth_date",
  "status": "missing",
  "value": null,
  "type": "date",
  "type_hint": "plain",
  "source_type": "end_user",
  "predefined_values": null,
  "title": "Title for date of birth birth date",
  "description": "Description for date of birth birth date",
  "style_hints":[]
}
```

**Comment by tomasperezv** [comment]
:+1:

**Pull request merged**